scalalib/src/mill/scalalib/RunModule.scala

```diff
@@ -46,27 +48,37 @@ trait RunModule extends WithZincWorker {
   def mainClass: T[Option[String]] = None
 
   /**
-   * Same as [[allLocalMainClasses]], but only for modules with a custom
-   * JVM version configured
-   */
-  def allLocalMainClasses0: T[Seq[String]] = Task { Seq.empty[String] }
-
-  /**
-   * All main classes detected in this module that can serve as program entrypoints
+   * All main classes detected in this module that can serve as program entry-points.
    */
   def allLocalMainClasses: T[Seq[String]] = Task {
-    if (zincWorker().javaHome().isEmpty) allLocalMainClasses0()
-    else {
-      Jvm.callProcess(
-        mainClass = "mill.scalalib.worker.DiscoverMainClassesMain",
-        classPath = zincWorker().classpath().map(_.path).toVector,
-        mainArgs = Seq(localRunClasspath().map(_.path).mkString(",")),
-        javaHome = zincWorker().javaHome().map(_.path),
-        stdin = os.Inherit,
-        stdout = os.Pipe,
-        cwd = Task.dest
-      ).out.lines()
+    classgraphWorker().discoverMainClasses(localRunClasspath().map(_.path))
+  }
+
+  def classgraphWorkerClasspath: T[Seq[PathRef]] = T {
+    zincWorker().defaultResolver().classpath(Seq(
+      Dep.millProjectModule("mill-scalalib-classgraph-worker")
+    ))
+  }
+
+  def classgraphWorker: Worker[ClassgraphWorker] = Task.Worker {
```

Comment:
Should this go on `ZincWorkerImpl`? It's mostly shared across all `RunModule`s, and doesn't seem to have any local configuration